lib/engine/index.ts

```diff
@@ -562,15 +562,15 @@ fn passSampleLevelBilinearRepeat(pass_index: int, uv: float2, lod: float) -> flo
             this.passF32
         );
 
-        if (this.bindings) {
-            // Copy over dynamic state
-            newBindings.custom = this.bindings.custom;
-            // newBindings.userData = this.bindings.userData;
-            newBindings.channels = this.bindings.channels;
-        }
+        // need deep clone here? it works somehow
+        // newBindings.custom = this.bindings.custom;
+        // newBindings.channels = this.bindings.channels;
+        // // newBindings.userData = this.bindings.userData;
```

Comment:
not sure why it works without deep clone, I don't noticed any issue.

Reply:
I should've tested before merging, this did in fact break (e.g. if you resize the window, it resets the engine and clears all the channel textures)